### PR TITLE
Feature parity for GraphQL queries between native-fungible and fungible

### PR DESCRIPTION
**Pull request by andresilva91**
## Motivation

Some `MapView` queries were missing, as indicated per https://github.com/linera-io/linera-protocol/issues/1798

## Proposal

Implement those missing queries

## Test Plan

Tested using GraphiQL, made sure new queries work and display expected results + also added coverage for these in the e2e tests

### CODE REVIEW

**Comment by andresilva91** [comment]
* **#1803** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1803?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @andresilva91 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: examples/native-fungible/src/service.rs [hunk 1]
```diff
@@ -61,8 +62,27 @@ impl Accounts {
...
+            .into_iter()
+            .map(|(owner, amount)| AccountEntry {
+                key: AccountOwner::User(owner),
+                value: amount,
+            })
+            .collect())
+    }
+
+    async fn keys(&self) -> Result<Vec<AccountOwner>, Error> {
+        Ok(system_api::current_owners()
```
Comment (afck) [inline examples/native-fungible/src/service.rs:82]: Maybe we should call it `current_balance_owners`, to distinguish it from the _chain_ owners.

**Review by MathieuDutSik — COMMENTED** [review]
Maybe I missed something, but the new features do not seem tested.

Otherwise, I find a little strange to have "map view" mentioned in the title. What is implemented is for "native-fungible". Whether that depends on MapView or not should have no importance. Maybe you want to say that now "native-fungible" is at feature parity with "fungible"?

**Review by jvff — APPROVED** [review]

File: linera-execution/src/lib.rs [hunk 2]
```diff
@@ -322,6 +322,12 @@ pub trait BaseRuntime {
     /// Reads the owner balance.
     fn read_owner_balance(&mut self, owner: Owner) -> Result<Amount, ExecutionError>;
 
+    /// Reads the balances from all owners.
+    fn read_owner_balances(&mut self) -> Result<Vec<(Owner, Amount)>, ExecutionError>;
+
+    /// Reads owners.
```
Comment (afck) [inline linera-execution/src/lib.rs:328]:
```suggestion
    /// Reads balance owners.
```

**Review by afck — APPROVED** [review]

File: linera-execution/src/execution_state_actor.rs [hunk 3]
```diff
@@ -109,6 +109,23 @@ where
...
+                        balances.push((owner, balance));
+                        Ok(())
+                    })
+                    .await?;
+                callback.respond(balances);
+            }
+
+            BalanceOwners { callback } => {
+                let balances = self.system.balances.indices().await?;
+                callback.respond(balances);
```
[the remark this answers is not in the document]
Comment (andresilva91) [reply linera-execution/src/execution_state_actor.rs]: Ha that was an oversight on my part, will fix

**Pull request merged**